File: crates/jvm-bindings/examples/ColorthiefTest.java
package colorthief;

import java.nio.file.Files;
import java.nio.file.Path;

public class ColorthiefTest {
    public static void main(String[] args) throws Exception {
        byte[] pixels = Files.readAllBytes(Path.of("examples/test.jpg"));
        // Note: This expects raw RGBA pixel data. Use Java AWT or TwelveMonkeys to decode.
        // byte[][] palette = Colorthief.getPalette(pixels, width, height, 10, 10);
        // System.out.println("Palette: " + java.util.Arrays.toString(palette));
        //
        // byte[] color = Colorthief.getColor(pixels, width, height, 10);
        // System.out.println("Dominant color: " + java.util.Arrays.toString(color));
    }
}

File: crates/jvm-bindings/examples/ColorthiefTest.kt
package colorthief

import java.nio.file.Files
import java.nio.file.Path

fun main() {
    val pixels = Files.readAllBytes(Path.of("examples/test.jpg"))
    // Note: This expects raw RGBA pixel data. Use Java AWT or TwelveMonkeys to decode.
    // val palette = Colorthief.getPalette(pixels, width, height, 10, 10)
    // println("Palette: ${palette.contentDeepToString()}")
    //
    // val color = Colorthief.getColor(pixels, width, height, 10)
    // println("Dominant color: ${color.contentToString()}")
}
